internal/namespaces/registry/v1/custom_namespace_login.go

```diff
@@ -0,0 +1,109 @@
+package registry
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"reflect"
+	"strings"
+
+	"github.com/scaleway/scaleway-cli/internal/core"
+	"github.com/scaleway/scaleway-sdk-go/api/registry/v1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+	"github.com/scaleway/scaleway-sdk-go/validation"
+)
+
+type registryLoginRequest struct {
+	Namespace string
+	Region    scw.Region
+	Program   string
+}
+
+func registryLoginCommand() *core.Command {
+	return &core.Command{
+		Short: `Login to a namespace`,
+		Long: `This command will run the correct command in order to log you in on the namespace with the chosen program.
+You will need to have the chosen binary installed on your system and in your PATH.`,
+		Namespace: "registry",
+		Resource:  "namespace",
+		Verb:      "login",
+		ArgsType:  reflect.TypeOf(registryLoginRequest{}),
+		ArgSpecs: []*core.ArgSpec{
+			{
+				Name:       "namespace",
+				Short:      `Namespace name or ID to log in to`,
+				Required:   true,
+				Positional: true,
+			},
+			{
+				Name:       "program",
+				Short:      "Program used to log in to the namespace",
+				Default:    core.DefaultValueSetter(string(docker)),
+				EnumValues: availablePrograms.StringArray(),
+			},
+			core.RegionArgSpec(),
+		},
+		Run: registryLoginRun,
+	}
+}
+
+func registryLoginRun(ctx context.Context, argsI interface{}) (i interface{}, e error) {
+	args := argsI.(*registryLoginRequest)
+
+	client := core.ExtractClient(ctx)
+	api := registry.NewAPI(client)
+
+	var endpoint string
+	if validation.IsUUID(args.Namespace) {
+		namespaceResp, err := api.GetNamespace(&registry.GetNamespaceRequest{
+			Region:      args.Region,
+			NamespaceID: args.Namespace,
+		})
+		if err != nil {
+			return nil, err
+		}
+		endpoint = namespaceResp.Endpoint
+	} else {
+		namespacesResp, err := api.ListNamespaces(&registry.ListNamespacesRequest{
+			Region: args.Region,
+			Name:   &args.Namespace,
+		}, scw.WithAllPages())
+		if err != nil {
+			return nil, err
+		}
+		if len(namespacesResp.Namespaces) == 0 {
+			return nil, fmt.Errorf("namespace not found")
+		} else if len(namespacesResp.Namespaces) > 1 {
+			return nil, fmt.Errorf("multiple namespaces match")
+		}
+		endpoint = namespacesResp.Namespaces[0].Endpoint
+	}
+	secretKey, ok := client.GetSecretKey()
+	if !ok {
+		return nil, fmt.Errorf("could not get secret key")
+	}
+
+	var commandArgs []string
+
+	switch program(args.Program) {
+	case docker, podman:
+		commandArgs = []string{"login", "-u scaleway", "--password-stdin", endpoint}
+	default:
+		return nil, fmt.Errorf("unknown program")
+	}
+
+	loginCommand := exec.Command(args.Program, commandArgs...)
+
+	loginCommand.Stdin = strings.NewReader(secretKey)
+	loginCommand.Stderr = os.Stderr
```

Comment:
why not binding stdout ?

Reply:
We should use streams that were passed to BoostrapConfig

Reply:
If we bind stoud we have the `Login Suceeded` message from docker